src/TraceEvent/Computers/TraceProcess.cs

```diff
@@ -322,7 +330,7 @@ internal TraceProcesses(TraceLog log, TraceEventDispatcher source)
         }
         internal TraceProcess GetOrCreateProcess(int processID, long timeQPC, bool isProcessStartEvent = false)
         {
-            Debug.Assert(processes.Count == processesByPID.Count);
+            Debug.Assert(source.DataLifetimeMsec > 0 /* lifetime tracking enabled */ || processes.Count == processesByPID.Count);
```

Comment:
Lets make a method DataLifetimeEnabled which is defined to be DataLifetimeMsec > 0 and use it here.   It seems useful in general to make code clearer.

Reply:
Done.